### PR TITLE
VTK Socks Setup for Windows (via WSL)

### PR DESCRIPTION
**Pull request by ericboehs**
This introduces support for Windows via WSL (Ubuntu).

Here's a pre-release video (final video I'll cut out access request tutorial):
<a href="https://www.youtube.com/watch?v=4ByTy_v-E8o">image</a>

- Removes HTTP connection testing as it would only fail on Mac. False positives!
- Now ensures private and public key match. If these keys mismatch a connection to the socks server will fail (I found this out the hard way). If no public key is found, SSH won't fail (weird, right?) so it skips the check.
- If you don't copy the key to the clipboard, it will display the key and the URL to submit your access form.
- In order to configure the underlying Windows machine, powershell.exe was used for copying to the clipboard, opening the browser, and setting the proxy.pac file in the Registry.
  - There is a bug in Windows where the font changes when powershell.exe is ran from Ubuntu. Re-opening the window fixes this
- If no credential helper is configured in git, a local cache of 10 minutes will be set. This is to prevent double prompting of user/pass on WSL (not a problem in regular ubuntu for some reason). It happens because I call `git clone` then `git checkout`. I couldn't think of a better solution.
- The ssh connection test now succeeds if the proxy is already running. It was hanging here on Windows (not sure how it worked on mac/linux before).
- If the SSH connection test fails for "Permission denied", it offers to copy your pub key to your clipboard as you may need to re-submit.
- In order to boot the proxy at Windows start, a very particular .bat file was created to launch the socks proxy via wsl. This causes a black console window to briefly open on boot. This could be alleviated with a .vbs script instead that would call the .bat file but I'd need to find another spot for the .bat file. Not a huge problem, just haven't solved it in this PR.
- The proxy_running command needs special tweaks for WSL as netstat and /proc/netstat doesn't see ports correctly.

### CODE REVIEW

File: lib/vtk/commands/socks/setup.rb [hunk 1]
```diff
@@ -40,8 +40,6 @@ def execute(input: $stdin, output: $stdout)
           configure_system_boot
           configure_system_proxy
 
-          test_http_connection unless skip_test
```
Comment (ericboehs) [inline lib/vtk/commands/socks/setup.rb:43]: This wasn't really testing. Basically could only fail on Mac. Removing for now.

File: lib/vtk/commands/socks/setup.rb [hunk 2]
```diff
@@ -456,7 +518,7 @@ def log(message)
 
             return_value = yield
 
-            output.puts return_value ? ' ✅' : ' ❌'
+            output.puts return_value ? ' ✅ DONE' : ' ❌ FAIL'
```
Comment (ericboehs) [inline lib/vtk/commands/socks/setup.rb:521]: I've add DONE/FAIL in all the places as WSL doesn't display emoji.

File: lib/vtk/commands/socks/setup.rb [hunk 3]
```diff
@@ -28,35 +28,49 @@ def initialize(options)
...
 
-          generate_key_and_open_key_access_request
+          @ssh_key_created = generate_key_and_open_key_access_request
         end
 
         def key_exists?
           File.exist? ssh_key_path
         end
 
+        def private_and_public_keys_match?
```
Comment (LindseySaari) [inline lib/vtk/commands/socks/setup.rb:63]: Nice! Great addition!

**Review by LindseySaari — APPROVED** [review]
🔥 ✅ Awesome Job!

**Pull request merged**